src/transport/raw/PeerAddress.h

```diff
@@ -133,18 +135,21 @@ class PeerAddress
     {
         char ip_addr[kInetMaxAddrLen];
 
+        char interface[Inet::InterfaceId::kMaxIfNameLength + 1];
+        mInterface.GetInterfaceName(interface, sizeof(interface));
+
         switch (mTransportType)
         {
         case Type::kUndefined:
             snprintf(buf, bufSize, "UNDEFINED");
             break;
         case Type::kUdp:
             mIPAddress.ToString(ip_addr);
-            snprintf(buf, bufSize, "UDP:%s:%d", ip_addr, mPort);
+            snprintf(buf, bufSize, "UDP:%s:%d %s", ip_addr, mPort, interface);
```

Comment:
I'll use '%' here

Reply:
The update is still not quite what I'm suggesting.

I'll post my suggested code change so it's clear.  Give me few minutes.

Reply:
This is what I'm suggesting.  I believe this would address @andy31415 's latest comment as well:

```
diff --git a/src/transport/raw/PeerAddress.h b/src/transport/raw/PeerAddress.h
index 3feac9991..f5ec8f601 100644
--- a/src/transport/raw/PeerAddress.h
+++ b/src/transport/raw/PeerAddress.h
@@ -136,8 +136,17 @@ public:
     {
         char ip_addr[kInetMaxAddrLen];
 
-        char interface[Inet::InterfaceId::kMaxIfNameLength + 1];
-        CHIP_ERROR err = mInterface.GetInterfaceName(interface, sizeof(interface));
+        char interface[Inet::InterfaceId::kMaxIfNameLength + 1] = {}; // +1 to prepend '%'
+        if (mInterface.IsPresent())
+        {
+            interface[0] = '%';
+            interface[1] = 0;
+            CHIP_ERROR err = mInterface.GetInterfaceName(interface + 1, sizeof(interface) - 1);
+            if (err != CHIP_NO_ERROR)
+            {
+                strncpy(interface, "%(err)", sizeof(interface));
+            }
+        }
 
         switch (mTransportType)
         {
@@ -146,11 +155,21 @@ public:
             break;
         case Type::kUdp:
             mIPAddress.ToString(ip_addr);
-            snprintf(buf, bufSize, "UDP:[%s]:%d%%%s", ip_addr, mPort, err == CHIP_NO_ERROR ? interface : "(err)");
+            #if INET_CONFIG_ENABLE_IPV4
+            if (mIPAddress.IsIPv4())
+                snprintf(buf, bufSize, "UDP:%s%s:%d", ip_addr, interface, mPort);
+            else
+            #endif
+                snprintf(buf, bufSize, "UDP:[%s%s]:%d", ip_addr, interface, mPort);
             break;
         case Type::kTcp:
             mIPAddress.ToString(ip_addr);
-            snprintf(buf, bufSize, "TCP:[%s]:%d%%%s", ip_addr, mPort, err == CHIP_NO_ERROR ? interface : "(err)");
+            #if INET_CONFIG_ENABLE_IPV4
+            if (mIPAddress.IsIPv4())
+                snprintf(buf, bufSize, "TCP:%s%s:%d", ip_addr, interface, mPort);
+            else
+            #endif
+                snprintf(buf, bufSize, "TCP:[%s%s]:%d", ip_addr, interface, mPort);
             break;
         case Type::kBle:
             // Note that BLE does not currently use any specific address.
```